pkg/filesystem/path/trace.go

```diff
@@ -44,3 +44,13 @@ func (t *Trace) GetUNIXString() string {
 	t.writeToStringBuilder(&sb)
 	return sb.String()
 }
+
+// GetWindowsString returns a string representation of the path for use on Windows.
+func (t *Trace) GetWindowsString() (string, error) {
+	if t == nil {
+		return ".", nil
+	}
+	var sb strings.Builder
+	t.writeToStringBuilder(&sb)
+	return sb.String(), nil
+}
```

Comment:
This still emits a UNIX style path containing potentially invalid filenames, etc. Probably best to rename `writeToStringBuilder` to `writeUNIXToStringBuilder` and add a `writeWindowsToStringBuilder`?

Reply:
Good. I am less familiar with the code paths for traces, I guess this should have an identity parsing test. Notably I do not think this tracks the drive letter as the code stands.

Reply:
Offline discussion: the Trace is only used for relative paths. So we do not need to worry about drive letters.


I have now moved the validation into the meat of the implementation. Will add a test.